pkg/endpoint/endpoint.go

```diff
@@ -1057,6 +1064,42 @@ func (e *Endpoint) SetK8sNamespace(name string) {
 	e.unlock()
 }
 
+// SetNamedPorts sets the k8s container ports specified by kubernetes.
+// Note that once put in place, the new k8sPorts is never changed,
+// so that the map can be used concurrently without keeping locks.
+// Reading the 'e.k8sPorts' member (the "map pointer") *itself* requires the endpoint lock!
+// Can't really error out as that might break backwards compatibility.
+func (e *Endpoint) SetNamedPorts(containerPorts []types.ContainerPort) error {
+	k8sPorts := make(policy.NamedPortsMap, len(containerPorts))
+	for _, cp := range containerPorts {
+		if cp.Name == "" {
+			continue // skip unnamed ports
+		}
+		if !api.IsNamedPort(cp.Name) {
+			log.WithField(logfields.PortName, cp.Name).Warning("ContainerPort: Invalid port name, not using as a named port")
```

Comment:
How can the user visualize this error better? Thinking on this better, this should never happened has the validation is done in k8s side.

Reply:
Right, this is likely not needed, but still wanted to have logging if this happens.